core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -203,7 +224,8 @@ private void rebalance(Chunk<K, V> c) {
         // will be redirected to help the rebalance procedure
         rebalancer.freeze();
 
-        rebalancer.createNewChunks(); // split or compact
+        ThreadContext ctx = getThreadLocalContext();
```

Comment:
(2) Since `rebalance()` is never called from outside, I agree it should receive the context as a parameter.
(1) This makes me rethink the idea of invalidating the context. Look at `put()` for example:
It gets the `ctx` at the beginning of the function, but it does not invalidate it after each retry.
So instead, I think it is better to rely on the methods that manage the context to invalidate it when necessary.
For example, the way `Chunk.lookup()` invalidates it when the key was not found.

Reply:
After looking at the code, (2) seems not a good idea. `rebalance()` is called from more often from methods that do not require the context object. So It will require the caller to get the context only for this method.